pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/SimpleLoadManagerImpl.java

```diff
@@ -1087,20 +1123,68 @@ public LoadReport generateLoadReport() throws Exception {
         if (timeSinceLastGenMillis <= LOAD_REPORT_UPDATE_MIMIMUM_INTERVAL) {
             return lastLoadReport;
         }
-
         try {
             LoadReport loadReport = new LoadReport(pulsar.getWebServiceAddress(), pulsar.getWebServiceAddressTls(),
-                pulsar.getBrokerServiceUrl(), pulsar.getBrokerServiceUrlTls());
-            loadReport.setName(String.format("%s:%s", pulsar.getAdvertisedAddress(), pulsar.getConfiguration().getWebServicePort()));
+                    pulsar.getBrokerServiceUrl(), pulsar.getBrokerServiceUrlTls());
+            loadReport.setName(String.format("%s:%s", pulsar.getAdvertisedAddress(),
+                    pulsar.getConfiguration().getWebServicePort()));
             SystemResourceUsage systemResourceUsage = this.getSystemResourceUsage();
             loadReport.setOverLoaded(
-                isAboveLoadLevel(systemResourceUsage, this.getLoadBalancerBrokerOverloadedThresholdPercentage()));
+                    isAboveLoadLevel(systemResourceUsage, this.getLoadBalancerBrokerOverloadedThresholdPercentage()));
             loadReport.setUnderLoaded(
-                isBelowLoadLevel(systemResourceUsage, this.getLoadBalancerBrokerUnderloadedThresholdPercentage()));
+                    isBelowLoadLevel(systemResourceUsage, this.getLoadBalancerBrokerUnderloadedThresholdPercentage()));
 
             loadReport.setSystemResourceUsage(systemResourceUsage);
             loadReport.setBundleStats(pulsar.getBrokerService().getBundleStats());
             loadReport.setTimestamp(System.currentTimeMillis());
+
+            final Set<String> oldBundles = lastLoadReport.getBundles();
+            final Set<String> newBundles = loadReport.getBundles();
+            final Set<String> bundleGains = new HashSet<>();
```

Comment:
is it possible to move this collection to class level and reuse it? as load-report run every x minutes and i may lil bit contribute in gc?